src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -91,6 +94,79 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
             return args.Count > 0 && args[0] is PythonCollection c ? c.Contents.FirstOrDefault() : null;
         }
 
+        public static IMember Super(IPythonModule declaringModule, IPythonFunctionOverload overload, IArgumentSet argSet, IndexSpan indexSpan) {
+            var args = argSet.Values<IMember>();
+
+            if (args.Count == 0) {
+                //Zero argument form only works inside a class definition
+                foreach (var s in argSet.Eval.CurrentScope.EnumerateTowardsGlobal) {
+                    if (s.Node is ClassDefinition) {
+                        var classType = s.Variables["__class__"].GetPythonType<IPythonClassType>();
+
+                        if (classType?.Mro != null) {
+                            return CreateSuper(argSet, classType.Mro, declaringModule, indexSpan);
+                        }
+                    }
+                }
+                return null;
+            }
+
+            // If multiple arguments first argument is required
+            var firstCls = args.FirstOrDefault().GetPythonType<IPythonClassType>();
+            if (firstCls == null) {
+                return null;
+            }
+
+            // second argument optional
+            bool isUnbound = args.Count == 1;
+            if (isUnbound) {
+                return CreateSuper(argSet, firstCls.Mro, declaringModule, indexSpan);
+            }
+
+            var secondCls = args[1].GetPythonType<IPythonClassType>();
+            if (secondCls != null &&
+                (secondCls.Equals(firstCls) || secondCls.IsSubClassOf(firstCls))) {
+                // We walk the mro of the second parameter looking for the first
+                return CreateSuper(argSet, secondCls.Mro, declaringModule, indexSpan, typeToFind: firstCls);
+            }
+
+            return null;
+        }
+
+        private static IMember CreateSuper(IArgumentSet argSet,
+            IReadOnlyList<IPythonType> callerMro,
+            IPythonModule declaringModule,
+            IndexSpan indexSpan,
+            IPythonType typeToFind = null) {
+
+            if (callerMro.Count == 0) {
+                return null;
+            }
+
+            IReadOnlyList<IPythonType> mro = callerMro;
+
+            // skip doing work if the newStartType is the first element in the callers mro
+            if (typeToFind != null &&
+                !typeToFind.Equals(callerMro.FirstOrDefault())) {
+                var mroList = callerMro.ToList();
```

Comment:
`typeToFind?.Equals(callerMro.FirstOrDefault()) == false` then no need to check for null. Basically `bool? == value`